homeassistant/components/zha/helpers.py

```diff
@@ -1116,7 +1116,14 @@ async def async_add_entities(
     if not entities:
         return
 
-    entities_to_add = [entity_class(entity_data) for entity_data in entities]
+    entities_to_add = []
+    for entity_data in entities:
+        try:
+            entities_to_add.append(entity_class(entity_data))
+        except Exception:  # noqa: BLE001
```

Comment:
Why is this a broad capturing? Can't we make this more specific?

Reply:
This was the least invasive way to catch any error from entity creation but I can definitely add try / catch to each `__init__` for all platforms and throw a much more specific error. Is that approach acceptible? I want to prevent a single misbehaving device / quirk from preventing an entire platforms entities from loading. Definitely open to suggestions.

Reply:
Sounds reasonable as a reason to do this; but was unclear. Maybe that should be added as a comment here in the code :)

Reply:
done

Reply:
need a test for coverage... will add in a bit

Reply:
Thanks 👍 

Will merge. I think for the future it should be researched how this can be prevented. Ideally, creating objects should not have side-effects (`__init__` should not be able to throw)...